### ISSUE TITLE
Trick S_define parser should return nonzero when it fails to find included header

### ISSUE THREAD

**Issue by abrogley** [nasa/trick#694]
I locally edited the SIM_Ball++_L1 sim by including a file that doesn't exist.

``` diff
abrogley@machine:SIM_Ball++_L1$ git diff S_define
diff --git a/trick_sims/SIM_Ball++_L1/S_define b/trick_sims/SIM_Ball++_L1/S_define
index da5f9d3b..26291121 100644
--- a/trick_sims/SIM_Ball++_L1/S_define
+++ b/trick_sims/SIM_Ball++_L1/S_define
@@ -10,6 +10,7 @@ LIBRARY DEPENDENCIES:
include "sim_objects/default_trick_sys.sm"

##include "Ball.hh"
+##include "path/does/not/exist.hh"
 
/**
This class is the base ball class
```

When the sim fails to build, I expected it to have a nonzero exit code, but it returns zero.

```
abrogley@machine:SIM_Ball++_L1$ ../../bin/trick-CP
Building with the following compilation flags:
TRICK_CFLAGS   =  -I./models
TRICK_CXXFLAGS =  -I./models
Running    configuration_processor
Parsing    S_define
Processing S_define
Processing /home/abrogley/trick/share/trick/sim_objects/default_trick_sys.sm
Processing SysSimObject
Processing MonteCarloSimObject
Processing MemoryManagerSimObject
Processing CheckPointRestartSimObject
Processing SieSimObject
Processing CommandLineArgumentsSimObject
Processing MessageSimObject
Processing JITSimObject
Processing InputProcessorSimObject
Processing ThreadProcessEventSimObject
Processing EventManagerSimObject
Processing VariableServerSimObject
Processing JSONVariableServerSimObject
Processing DataRecordDispatcherSimObject
Processing RTSyncSimObject
Processing FrameLogSimObject
Processing MasterSlaveSimObject
Processing InstrumentationSimObject
Processing DMTCPSimObject
Processing InjectorExecSimObject
Processing InjectorSimObject
Processing ZeroconfSimObject
Processing UnitTestSimObject
Processing UdUnitsSimObject
Continuing S_define
Error      Could not find included file "path/does/not/exist.hh"
Trick Build Process Complete
make: `all' is up to date.
abrogley@machine:SIM_Ball++_L1$ echo $?
0
```

This behavior also happens if there is a syntax error in the S_define.

**Comment by abrogley** [comment]
Commit ca99386 did not fix this issue for me on Ubuntu 16.04 or MacOS 10.14.1

**Comment by ddj116** [comment]
I just ran into this in ramtares using Trick 17.5.0 on CentOS 7.6 (FSL).  This is a **huge deal** for CI -- if we can't trust the return code of `trick-CP` our entire CI system is useless!  Can we accelerate a fix for this?

**Comment by dbankieris** [comment]
I believe this was fixed as a secondary effect of b3c9e75. The above example correctly returns a non-zero exit code on `master`.

**Comment by ddj116** [comment]
I'll update to the latest trick version and see if my example clears. Will report back.

**Comment by ddj116** [comment]
Can confirm this is fixed on origin/master (adc3a3e425235c3dcf57c047899be46d4e4df1d7) but not the latest tagged release 17.5.0.
```
Processing InjectorSimObject
Processing ZeroconfSimObject
Processing UnitTestSimObject
Processing UdUnitsSimObject
Continuing S_define
Error      Could not find included file "ball/L1/include/ball_yo.h"
make: *** [S_source.hh] Error 255
user@blade04-fsl: /scratch/00/user/ramtares/trick/trick_sims/Ball/SIM_ball_L1 > echo $?
2
```
We will update to this latest commit and move foward.  Thanks @dbankieris !
